index.bs

```diff
@@ -499,6 +523,64 @@ Modify the <a spec=HTML>downloads a hyperlink</a> algorithm to ensure that downl
   1. If [=this=]'s [=Window/browsing context=] is a [=prerendering browsing context=], then return.
 </div>
 
+<h3 id="permissions">Waiting before permissions checks</h3>
+
+Many specifications need to be patched so that, if a given algorithm invoked in a [=prerendering browsing context=], most of its work is deferred until the browsing context is [=prerendering browsing context/activated=]. This is tricky to do uniformly, as many of these specifications do not have great hygeine around <a href="https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-for-spec-authors">using the event loop</a>. Nevertheless, the following sections give our best attempt.
+
+<h4 id="patch-geolocation">Geolocation API</h4>
+
+<div algorithm="Geolocation getCurrentPosition patch">
+  Modify the {{Geolocation/getCurrentPosition()}} method steps by prepending the following step:
+
+  1. If [=this=]'s [=relevant global object=]'s [=Window/browsing context=] is a [=prerendering browsing context=], then append the following steps to [=this=]'s [=platform object/post-prerendering activation steps list=] and return.
```

Comment:
I wonder if we'd want to simply define a shorthand like "continue after prerendering" and use it similarly to how upstream geolocation API says "continue asynchronously". Is there some better spec analogue for async/await?

Reply:
Yeah, I'm thinking that might be a good idea. I want to get 3-4 more long-handed samples done before doing such a refactoring, but ultimately this should be as easy as possible to use from other specs, and providing a macro of that sort seems like the right method.